pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -459,13 +578,22 @@ func (p *staticPolicy) guaranteedCPUs(pod *v1.Pod, container *v1.Container) int
 		return 0
 	}
 	cpuQuantity := container.Resources.Requests[v1.ResourceCPU]
-	// In-place pod resize feature makes Container.Resources field mutable for CPU & memory.
-	// AllocatedResources holds the value of Container.Resources.Requests when the pod was admitted.
-	// We should return this value because this is what kubelet agreed to allocate for the container
-	// and the value configured with runtime.
 	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
-		if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
-			cpuQuantity = cs.AllocatedResources[v1.ResourceCPU]
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScalingExclusiveCPUs) {
+			// TODO Fix InPlacePodVerticalScaling alognside CPU Static policy, until fixed
+			// we return AllocatedResources value because this is what kubelet agreed to allocate for the container
+			// and the value configured with runtime.
+			klog.InfoS("InPlacePodVerticalScaling alognside CPU Static policy not implemented, reverting to default behaviour disallowing pod resize")
+			if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+				cpuQuantity = cs.AllocatedResources[v1.ResourceCPU]
```

Comment:
@esotsal, Hi, Mr. escotal, can you check this part of code "cpuQuantity = cs.AllocatedResources[v1.ResourceCPU]"

When test InplacePodVerticalScaling for guaranteed Pod based on this code, and the phenomenon is the InplacePodVerticalScaling resize result is fall behind the command want to be. But change cpuQuantity to be "cpuQuantity := container.Resources.Requests[v1.ResourceCPU]", the Pod can be scaled successful.
So I think the cpuQuantity := container.Resources.Requests[v1.ResourceCPU] is right, and cpuQuantity = cs.AllocatedResources[v1.ResourceCPU] may need more consider.

The test command and result as below:
# kubectl patch pod new-pod-0 --subresource resize --patch '{"spec":{"containers":[{"name":"new-pod-container-0", "resources":{"requests":{"cpu":"5"}, "limits":{"cpu":"5"}}}]}}'
pod/new-pod-0 patched

# sudo cat /var/lib/kubelet/cpu_manager_state
{"policyName":"static","defaultCpuSet":"0,8-10,17-19","entries":{"9d07a1a2-09af-49f4-bb2d-b51421cad78f":{"c-111111":"1"},"c69cf73a-f6e2-4b8c-b62c-f32fd915c308":{"new-pod-container-0":"2-7,12-16"},"ed17948b-d8ee-47fc-8402-a911945cc309":{"c-222222":"11"}},"checksum":268222143}

# kubectl patch pod new-pod-0 --subresource resize --patch '{"spec":{"containers":[{"name":"new-pod-container-0", "resources":{"requests":{"cpu":"10"}, "limits":{"cpu":"10"}}}]}}'
pod/new-pod-0 patched

# sudo cat /var/lib/kubelet/cpu_manager_state
{"policyName":"static","defaultCpuSet":"0,4-6,8-10,14-19","entries":{"9d07a1a2-09af-49f4-bb2d-b51421cad78f":{"c-111111":"1"},"c69cf73a-f6e2-4b8c-b62c-f32fd915c308":{"new-pod-container-0":"2-3,7,12-13"},"ed17948b-d8ee-47fc-8402-a911945cc309":{"c-222222":"11"}},"checksum":1668433879}

# kubectl patch pod new-pod-0 --subresource resize --patch '{"spec":{"containers":[{"name":"new-pod-container-0", "resources":{"requests":{"cpu":"3"}, "limits":{"cpu":"3"}}}]}}'
pod/new-pod-0 patched

# sudo cat /var/lib/kubelet/cpu_manager_state
{"policyName":"static","defaultCpuSet":"0,6,8-10,16,18-19","entries":{"9d07a1a2-09af-49f4-bb2d-b51421cad78f":{"c-111111":"1"},"c69cf73a-f6e2-4b8c-b62c-f32fd915c308":{"new-pod-container-0":"2-5,7,12-15,17"},"ed17948b-d8ee-47fc-8402-a911945cc309":{"c-222222":"11"}},"checksum":2548566557}

Reply:
Hi, thanks I will take a look when i am back from vacation second week of January.